server_handlers.go

```diff
@@ -111,8 +120,20 @@ func serverHandshakeHandler(c *Conn) error {
 				} else if !bytes.Equal(expectedVerifyData, h.verifyData) {
 					return errVerifyDataMismatch
 				}
+
+				if c.clientAuth == RequireAnyClientCert && c.remoteCertificate == nil {
+					return errClientCertificateRequired
+				} else if c.remoteCertificate != nil && !c.remoteCertificateVerified {
+					return errClientCertificateNotVerified
+				}
+
+				if c.clientAuth > NoClientCert {
+					c.localSequenceNumber = 6
```

Comment:
Is this something in the spec? Or are we incrementing the `localSequenceNumber` for another part of the code?

Reply:
Incrementing for another part of the code.

We only increment when we move a flight, because the connection is lossy we keep replaying and it makes it easier.

What would be nice is to have a function `baseSequenceNumberForFlight` or something like that, would probably be better then me just setting them manually.